packages/amplication-client/src/Project/ArchitectureConsole/hooks/useModelOrganizer.tsx

```diff
@@ -577,11 +583,25 @@ const useModelOrganization = ({ projectId, onMessage }: Props) => {
           (x) => x.entityId !== node.id
         );
 
+        const authEntity =
+          resourceSettings?.serviceSettings?.authEntityName ===
+          currentNode.data.payload.name;
+
         if (
-          duplicatedEntityName &&
+          (duplicatedEntityName || authEntity) &&
           currentNode.data.originalParentNode !== currentNode.parentNode
         ) {
+          const baseErrorMessage = `Cannot move entity to service: ${currentTargetResource.data.payload?.name}`;
           currentNode.parentNode = currentNode.data.originalParentNode;
+          if (authEntity) {
+            setDuplicateEntityErrorMessage(
+              `${baseErrorMessage} because the authentication entity cannot be deleted.`
+            );
+          } else {
+            setDuplicateEntityErrorMessage(
+              `${baseErrorMessage} because the entity name already exists.`
```

Comment:
can we use a logic a head of time and prevent the node from being moved? 
we can use the node.draggable= false 
we can keep this logic anyway if needed